proposals/0000-linear-types.rst

```diff
@@ -0,0 +1,858 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+.. highlight:: haskell
+
+This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_. **After creating the pull request, edit this file again, update the number in the link, and delete this bold sentence.**
+
+.. contents::
+
+Linear Types
+============
+
+This proposal introduces a notion of *linear function* to GHC. Linear
+functions are regular functions that guarantee that they will use
+their argument exactly once. Whether a function ``f`` is linear or not
+is called the *multiplicity* of ``f``. We propose a new language
+extension, ``-XLinearTypes``. When turned on, the user can enforce
+a given multiplicity for ``f`` using a type annotation.
+
+The theory behind this proposal has been fully developed in a peer
+reviewed conference publication that will be presented at POPL'18. See
+the `extended version of the paper
+<https://arxiv.org/abs/1710.09756>`_.
+
+Motivation
+----------
```

Comment:
This section is too long and parochial. Really, linear types can model any resource which is not duplicable. This is in no way limited to memory or IO-things. There are some benefits to modeling (EDSL) which are not touched upon here. (Networking, information flow, etc.)

The last sentence goes on to say: "plus other unknown things"... but really there is a wealth of literature on applications.